src/transforms/aws_ec2_metadata.rs

```diff
@@ -542,32 +562,44 @@ mod integration_tests {
         sleep(Duration::from_secs(1)).await;
 
         let log = LogEvent::default();
-        let mut expected = log.clone();
+        let metric = make_metric();
+        let mut expected_log = log.clone();
+        let mut expected_metric = metric.clone();
```

Comment:
How hard would it be to move the log and metric tests into separate `#[test]` functions? That way if only one fails it is immediately obvious which code path is involved.